src/main/java/com/rocketden/main/util/Utility.java

```diff
@@ -7,13 +7,24 @@
 
 import com.rocketden.main.dao.RoomRepository;
 import com.rocketden.main.dao.UserRepository;
+import com.rocketden.main.game_object.NotificationType;
 
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Component;
 
 @Component
 public class Utility {
 
+    // List notifications that require an initiator.
+    public static final List<NotificationType> initiatorNotifications =
+        Arrays.asList(NotificationType.SUBMIT_CORRECT,
+            NotificationType.SUBMIT_INCORRECT, NotificationType.CODE_STREAK,
+            NotificationType.TEST_CORRECT);
+
+    public static final List<NotificationType> contentNotifications = 
+        Arrays.asList(NotificationType.SUBMIT_CORRECT,
+            NotificationType.TEST_CORRECT, NotificationType.TIME_LEFT);
```

Comment:
One thing I wanted to note is I feel like all of these notifications save `CODE_STREAK` would be generated on the backend (also to prevent random users from just spamming any notification they want) - that might also make it easier cause we wouldn't really need to worry about initiatorNotifications and contentNotifications and differentiating them with different API errors

This is just me jotting down my thoughts tho (feel free to add or refute or anything) - not sure if I'm really requesting any changes here

Reply:
This is all fair, I'm mostly checking for things that we control ourselves in the code, so there's an argument to be made for "why check that". I figure the extra security can't hurt, but it may if it is excessive, so I'll leave any changes on this up to you -- I'm fine with leaving it or removing it.

Reply:
Sounds good - I see no harm in leaving it for now so all good